### PR TITLE
Consult the council

### PR DESCRIPTION
**Pull request by sylvainsf**
# Description

Add Rivendell Council as an AI-powered code review workflow for pull requests. This integrates [sylvainsf/rivendell_council](https://github.com/sylvainsf/rivendell_council) as a reusable GitHub Actions workflow that convenes a fellowship of AI reviewers — each with a specialized domain — to evaluate every PR.

The council reviews PRs in parallel using [GitHub Models](https://github.com/marketplace/models) via `actions/ai-inference`, requiring no external API keys. Each member scores the diff on a 1–10 scale and provides concerns and suggestions. Gandalf synthesizes all reviews and delivers a final PASS/FAIL verdict as a PR comment.

**Council members:**
- 🌟 **Elrond** — Correctness (logic, edge cases, tests)
- ⚔️ **Gimli** — Strength (security, validation, error handling)
- 👑 **Aragorn** — Maintainability (structure, idioms, naming)
- 🏹 **Legolas** — Performance (efficiency, resource usage)
- 🧙 **Gandalf** — Arbiter (synthesizes all reviews, delivers verdict)

**Changes:**
- Added `.github/workflows/council-review.yml` calling the reusable workflow

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: N/A

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- An overview of proposed schema changes is included in a linked GitHub issue.
    - [ ] Yes 
    - [x] Not applicable 
- A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
    - [ ] Yes 
    - [x] Not applicable 
- The design document has been reviewed and approved by Radius maintainers/approvers.
    - [ ] Yes 
    - [x] Not applicable 
- A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
    - [ ] Yes 
    - [x] Not applicable 
- A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
    - [ ] Yes 
    - [x] Not applicable 
- A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.
    - [ ] Yes 
    - [x] Not applicable

### CODE REVIEW

File: .github/workflows/council-review.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Rivendell Council Review
```
[the remark this answers is not in the document]
Comment (sylvainsf) [reply .github/workflows/council-review.yml:3]: @copilot apply changes based on [this feedback](https://github.com/radius-project/radius/pull/11432#discussion_r2937370954)

File: .github/workflows/council-review.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    types: [opened, synchronize, reopened]
+
+jobs:
+  council:
+    name: Convene the Council
+    uses: sylvainsf/rivendell_council/.github/workflows/council.yml@main
+    permissions:
+      contents: read
+      pull-requests: write
+      models: read
```
[the remark this answers is not in the document]
Comment (sylvainsf) [reply .github/workflows/council-review.yml:20]: @copilot apply this fix as well

File: .github/workflows/council-review.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    types: [opened, synchronize, reopened]
+
+permissions: {}
+
+jobs:
+  council:
+    name: Convene the Council
+    # Skip on fork PRs - GITHUB_TOKEN is read-only for forks and cannot write PR comments
+    if: github.event.pull_request.head.repo.full_name == github.repository
+    uses: sylvainsf/rivendell_council/.github/workflows/council.yml@v1
```
Comment (DariuszPorowski) [inline .github/workflows/council-review.yml:16]: issue: action must be pin to sha - it's a repo level enforcement, anything else will fail.

File: .github/workflows/council-review.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    permissions:
+      contents: read
+      pull-requests: write
+      models: read
+    with:
+      model: openai/gpt-4o
```
Comment (DariuszPorowski) [inline .github/workflows/council-review.yml:22]: idea: maybe a bit newer model? like gpt-5?

**Review by DariuszPorowski — CHANGES_REQUESTED** [review]
